CHANGELOG.md

```diff
@@ -2,23 +2,190 @@
+
+    Added in: [#30](https://github.com/Plutonomicon/plutarch/pull/30)
+  - `PEq` instance for `PData` and `PAsData`
+
+    Module: `Plutarch.Builtin`
+
+    Added in: [#38](https://github.com/Plutonomicon/plutarch/pull/38)
+  - Tracing functions:- `ptrace`, `ptraceIfFalse`, `ptraceIfTrue`, `ptraceError`
+
+    Module: `Plutarch.Trace`
+
+    Added in: [#39](https://github.com/Plutonomicon/plutarch/pull/39)
+  - Builtin pair construction utility:- `ppairDataBuiltin `
+
+    Module: `Plutarch.Builtin`
+
+    Added in: [#50](https://github.com/Plutonomicon/plutarch/pull/50)
+  - *Loads* of awesome list utilities thanks to [#63](https://github.com/Plutonomicon/plutarch/pull/63)!
+
+    Module: `Plutarch.List`
+  - `PIsData` instance for `PBool`
+
+    Module: `Plutarch.Builtin`
+
+    Added in: [#110](https://github.com/Plutonomicon/plutarch/pull/110)
+
+  </details>
+- Add `PIsData` for conversion between normal builtin types and their `Data` representation.
+
+  Initially added in: [#31](https://github.com/Plutonomicon/plutarch/pull/31)
+- Add `PAsData` for preserving more type information regarding `Data` encoded values.
+
+  Initially added in: [#31](https://github.com/Plutonomicon/plutarch/pull/31)
+- Add `PDataRepr` and related machinery to ergonomically work with `Constr` encoded data.
+
+  There are also **generic derivers** (!!) to implement the related typeclasses for custom data types.
+
+  Module: `Plutarch.DataRepr`
+
+  Worked on through
+  * [#31](https://github.com/Plutonomicon/plutarch/pull/31)
+  * [#169](https://github.com/Plutonomicon/plutarch/pull/169)
+  * [#173](https://github.com/Plutonomicon/plutarch/pull/173)
+  * [#176](https://github.com/Plutonomicon/plutarch/pull/176)
+  * [#171](https://github.com/Plutonomicon/plutarch/pull/171)
+- Add `PLift` and `PConstant` - enabling conversion between Plutarch terms and Haskell types.
+
+  This comes with convenient derivers. See the relevant section on the guide.
+
+  Module: `Plutarch.Lift`; another deriver within `Plutarch.DataRepr.Internal`
+
+  Worked on through
+  * [#62](https://github.com/Plutonomicon/plutarch/pull/62)
+  * [#109](https://github.com/Plutonomicon/plutarch/pull/109)
+  * [#130](https://github.com/Plutonomicon/plutarch/pull/130)
+- Deprecate `punsafeConstant`. Use `pconstant` instead!
+- Add `PIsList` & `PList` + instances for `PBuiltinList` - ergonomic list functions for all!
+
+  Module: `Plutarch.List`; another instance in `Plutarch.Builtin`
+
+  Added in: [#63](https://github.com/Plutonomicon/plutarch/pull/63)
+- Mutually recursive data types, scott encoded records and more provided by `Plutarch.Rec`!
+
+  Module: `Plutarch.Rec`
+
+  Worked on through:
+  * [#60](https://github.com/Plutonomicon/plutarch/pull/60)
+  * [#114](https://github.com/Plutonomicon/plutarch/pull/114)
+  * [#125](https://github.com/Plutonomicon/plutarch/pull/125)
+  * [#175](https://github.com/Plutonomicon/plutarch/pull/175)
+- The Plutus V1 ledger api implemented in Plutarch!
```

Comment:
Should probably say 'api types' - since we don't have functions yet.

Reply:
Agreed, updated!